docs/Configuring-Orion/TLS.md

```diff
@@ -3,15 +3,162 @@ description: TLS
+
+The following procedures explain how to generate certificates when the Common Name (CN) is either the public DNS or an IP address by using [OpenSSL](https://www.openssl.org/source/). Before you begin, ensure OpenSSL is installed.
+
+Choose one of the following methods:
+
+* [Generating certificates when the public DNS is the CN](#Generating-certificates-when-the-public-DNS-is-the-CN)
+* [Generating certificates when the public IP address is the CN](#Generating-certificates-when-the-public-IP-address-is-the-CN)
+
+### Generating certificates when the public DNS is the CN
+
+If you want to use a public DNS as the CN, complete the following steps:
+
+#### Generating a CA certificate
+
+1. Generate a key file called orion_ca.key by running the following command:
+
+    `openssl genrsa -out orion_ca.key 2048`
+  
+1. Generate a CA certificate called orion_ca.pem that uses orion_ca.key by running the follow command:
+
+    `openssl req -x509 -new -nodes -key orion_ca.key -sha256 -days 1024 -out orion_ca.pem`
+
+#### Generating a new certificate for a node
+
+We recommend each node has its own certificate. To generate the certificate, complete the following process:
+
+1. Generate a key file called orion_cer.key by running the following command:
+
+    `openssl genrsa -out orion_cer.key 2048`
+  
+1. Generate a certificate signing request (CSR) called orion_cer.csr by running the following command:
+
+    `openssl req -new -key orion_cer.key -out orion_cer.csr`
+  
+1. Answer each prompt for information that will be added to the certificate request. Ensure that the
+value you specify for Common Name (CN) matches the host public DNS so that the requests from the server are accepted. The name 
+is used in the configuration file for the `nodeurl` and `clienturl` options too.
+
+1. Generate a certificate called orion_cer.pem signed by the CA certificate by runnig the following command:
+
+    `openssl x509 -req -in orion_cer.csr -CA orion_ca.pem -CAkey orion_ca.key -CAcreateserial -out orion_cer.pem -days 500 -sha256`
+
+#### Configuration file example that uses TLS and the created certificates
+
+```
+nodeurl = "<YOUR-COMMON-NAME>:8080/"
+nodeport = 8080
+
+clienturl = "<YOUR-COMMON-NAME>:8888/"
+clientport = 8888
+
+othernodes = ["<YOUR-COMMON-NAME>:8080"]
+
+workdir = "<PATH-TO>/orion/bin"
+
+publickeys = ["orion.pub"]
+privatekeys = ["orion.key"]
+
+tls = "strict"
+
+tlsservercert = "<PATH-TO>/orion/bin/orion_cer.pem"
+tlsserverchain = ["<PATH-TO>/orion/bin/orion_ca.pem"]
+tlsserverkey = "<PATH-TO>/orion/bin/orion_cer.key"
+
+tlsclientcert = "<PATH-TO>/orion/bin/orion_cer.pem"
+tlsclientchain = ["<PATH-TO>/orion/bin/orion_ca.pem"]
+tlsclientkey = "<PATH-TO>/orion/bin/orion_cer.key"
+
+tlsclienttrust = "ca"
+tlsservertrust = "ca"
+
+nodenetworkinterface = "0.0.0.0"
+clientnetworkinterface = "0.0.0.0"
+```
+
+### Generating certificates when the public IP address is the Common Name (CN) 
```

Comment:
I won't repeat the same comments on this section but the comments from above will also need to be applied to this section.